### ISSUE TITLE
Issue: Unable to add a local SpotWeb instance as an Indexer

### ISSUE THREAD

**Issue by heymoe** [theotherp/nzbhydra2#102]
Version: NZBHydra2 v1.2.3

Trying to add a local SpotWeb instance as an Indexer:
https://github.com/spotweb/spotweb

which can be configured as a newznab provider:
https://github.com/spotweb/spotweb/wiki/Spotweb-als-Newznab-Provider

Seems like it will not activate it until capabilities checks are done. The local SpotWeb instance does not require an API key but the check capabilities button does not show up unless there is an API key provided.

If I add an API key and click the check capabilities button it just fails and says it won't activate the Indexer until the checks run.

Adding the local SpotWeb instance as an Indexer in NZBHydra (not v2) works fine and it doesn't require an API key to be entered and activates it without the need to run the capabilities checks.

The settings for the SpotWeb Indexer are the same between the NZBHydra v1 and v2 instances. 

My guess is that the SpotWeb newznab emulation isn't 100% and thus the failing the checks and/or sending the fake API key doesn't make it happy since it doesn't need one.

**Comment by theotherp** [comment]
There's no need to click the caps check button. On clicking OK it will be executed automatically.

I used to hide the caps check button and search type/ID input fields when a new indexer is added but that didn't sit right with others for some reason.

Have you tried just clicking OK?

**Comment by heymoe** [comment]
I did try just clicking Submit / OK and I get:


The connection to the indexer failed: Error while communicating with indexer SpotWEB. Server returned: org.springframework.oxm.UnmarshallingFailureException: JAXB unmarshalling exception; nested exception is javax.xml.bind.UnmarshalException - with linked exception: [org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog.]
Do you want to add it anyway?


and I get 3 options:
```
"I know what I'm doing"
"Add it, but disabled"
"Ahhh, let me try again"
```

I tried clicking on "I know what I'm doing" but it still doesn't work but going back to Hydra v1, it's fine there. 

The SpotWeb Indexer does show up as an selection on the Search page but when you try to use it (select it / check it) and try a search it says:

```
No indexers were picked for this search
Reasons:
SpotWeb: Configuration incomplete
```

**Comment by heymoe** [comment]
I have added 4 other non SpotWeb indexers and those all work just fine in Hydra v2

**Comment by theotherp** [comment]
I'll need to run a spotweb instance on my own to test this. Will have to wait until the weekend.

**Comment by heymoe** [comment]
No worries.. Let me know if there is any other testing I can do on my side.

**Comment by theotherp** [comment]
I just remembered 6box has a spotweb instance which allowed me to test it. Please let me know if the fix works for you.

**Comment by heymoe** [comment]
Upgraded to v1.2.4 and adding the SpotWeb Indexer works like a charm now. It successfully does the compatibility checks (as in detects the supported Search IDs and Search Types) without errors but when I try to do a search using it I get this:

Error while communicating with indexer SpotWEB. Server returned: org.springframework.oxm.UnmarshallingFailureException: JAXB unmarshalling exception; nested exception is javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"html"). Expected elements are <{}caps>,<{}categories>,<{}category>,<{}error>,<{}guid>,<{}item>,<{}limits>,<{http://www.newznab.com/DTD/2010/feeds/attributes/}response>,<{}rss>,<{}searching>,<{}xml>

**Comment by theotherp** [comment]
Yeah, because I'm an idiot.

**Comment by theotherp** [comment]
Fixed now.

**Comment by heymoe** [comment]
v1.2.5 working great now. Thanks!
